qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -1,15 +1,76 @@
 from functools import partial
 import numpy as np
-from typing import Any
+from typing import Any, Union
 
 from qcodes import VisaInstrument
-from qcodes.instrument.parameter import ArrayParameter, ParamRawDataType
-from qcodes.utils.validators import Numbers, Ints, Enum, Strings
+from qcodes.instrument.parameter import ArrayParameter, ParamRawDataType, ParameterWithSetpoints, Parameter
+from qcodes.utils.validators import Numbers, Ints, Enum, Strings, Arrays
 
 from typing import Tuple
 import time
 
 
+class ChannelTrace(ParameterWithSetpoints):
+    """
+    Parameter class for the two channel buffers
+    """
+
+    def __init__(self, name: str, channel: int, **kwargs: Any) -> None:
+        """
+        Args:
+            name: The name of the parameter
+            channel: The relevant channel (1 or 2). The name should
+                should match this.
+        """
+        super().__init__(name, **kwargs)
+
+        self._valid_channels = (1, 2)
+
+        if channel not in self._valid_channels:
+            raise ValueError('Invalid channel specifier. SR830 only has '
+                             'channels 1 and 2.')
+
+        if not isinstance(self.root_instrument, SR830):
+            raise ValueError('Invalid parent instrument. ChannelBuffer '
+                             'can only live on an SR830.')
+
+        self.channel = channel
+        self.update_unit()
+
+    def update_unit(self) -> None:
+        assert isinstance(self.root_instrument, SR830)
+        params = self.root_instrument.parameters
+        if params[f'ch{self.channel}_ratio'].get() != 'none':
+            self.unit = '%'
+        else:
+            disp = params[f'ch{self.channel}_display'].get()
+            if disp == 'Phase':
+                self.unit = 'deg'
+            else:
+                self.unit = 'V'
+                self.label = self.root_instrument.name + '_' + disp
```

Comment:
> Using the instrument in the label
The Scientist often uses more, than one SR830 at a time. Thus, having the instrument  as part of  the label helps the user to instantly decode the plot. It was after a request from the people I was helping, but maybe it is to focused one there use case, and it would be better to just give people the change to supply a custom label.

> should the channel number be also part of the label:
The channel can be inferred from “disp”.  The instrument does not really have channels but two displays, so including channels in the label would be confusing for someone who have not read the driver.  The two displays can not show the same output.

Reply:
thanks for the explanation! and i support this:

> it would be better to just give people the change to supply a custom label